### PR TITLE
Add  c100-shared-infrastructure.json

### PR DESCRIPTION
**Pull request by Ramparitala**
### Change description ###
Added c100-shared-infrastructure.json

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Review by timja — COMMENTED** [review]
subnets shouldn't be created in repositories generally they are managed centrally,

can you provide more context please?

what's the private endpoint for? generally those should be created in side terraform modules

**Comment by Ramparitala** [comment]
@timja  I am trying to create resources (Postgress and Redis in sandbox subscription). While running the pipeline I got this whitelisted resources error. So, I have taken the reference of one of our service help-with-fees and did the same change. Please point me to the direction if I am doing something irrelevant.  

HWF reference: 
https://github.com/hmcts/sds-jenkins-config/commit/1e5e4b4982ee2952514b8a937b5ba3dfe6686d2e  

Job info: 
https://sds-build.platform.hmcts.net/job/HMCTS_C100/job/c100-shared-infrastructure/job/demo/lastFailedBuild/console

**Comment by timja** [comment]
`azurerm_private_endpoint` is okay for PET because there's a networking issue stopping you from accessing the default private endpoint in SDS, but why do you need the subnet one?

**Comment by Ramparitala** [comment]
@timja Here we are not trying to create another subnet it is just trying to retrieve the existing subnet information to create Postgress and Redis.

**Comment by Ramparitala** [comment]
@timja Here we are not trying to create another subnet it is just trying to retrieve the existing subnet information to create Postgress and Redis.

**Comment by timja** [comment]
data resources are always allowed no need to include them here

File: terraform-infra-approvals/c100-shared-infrastructure.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "resources": [
+    {"type": "azurerm_subnet"},
```
Comment (timja) [inline terraform-infra-approvals/c100-shared-infrastructure.json:3]:
```suggestion
```

**Review by timja — APPROVED** [review]

**Pull request merged**